language/move-vm/runtime/src/loader.rs

```diff
@@ -681,6 +686,16 @@ impl Loader {
         check_natives_impl(module).map_err(|e| e.finish(Location::Module(module.self_id())))
     }
 
+    // We do not allow cyclic dependencies between modules, i.e., module A depends on module B
+    // while at the same time, B (transitively) depends on A.
+    fn verify_no_cyclic_module_dependencies(&self, module: &CompiledModule) -> VMResult<()> {
+        CyclicModuleDependencyChecker::verify_module(module, |module_id| {
+            self.get_module(module_id)
+                .module()
+                .immediate_module_dependencies()
+        })
+    }
+
```

Comment:
I'd remove this function. It should really be a part of `verify_module_dependencies`